internal/converter/internal/staticconvert/internal/build/converter_remotewriteexporter.go

```diff
@@ -91,13 +91,20 @@ func toremotewriteexporterConfig(cfg *remotewriteexporter.Config, forwardTo []st
 	defaultArgs := &prometheus.Arguments{}
 	defaultArgs.SetToDefault()
 
-	return &prometheus.Arguments{
+	args := &prometheus.Arguments{
 		IncludeTargetInfo:             defaultArgs.IncludeTargetInfo,
 		IncludeScopeInfo:              defaultArgs.IncludeScopeInfo,
 		IncludeScopeLabels:            defaultArgs.IncludeScopeLabels,
-		GCFrequency:                   cfg.StaleTime,
+		GCFrequency:                   defaultArgs.GCFrequency,
 		ForwardTo:                     forwardTo,
 		AddMetricSuffixes:             defaultArgs.AddMetricSuffixes,
 		ResourceToTelemetryConversion: defaultArgs.ResourceToTelemetryConversion,
 	}
+
+	// Override default only if > 0 because GCFrequency of 0 is not allowed
```

Comment:
Same here, do we really need to check if the incoming config is invalid? Also, if we set a different GC frequency than what the user specified, we're technically not converting their exact config anymore. Maybe it'd make more sense to just refuse to convert the config, or to log a warning.

But IMO we could just convert it as it is and let Alloy fail when it loads it.

Reply:
I dug deeper into this one because I could not understand why it would be set to 0 by default and I found out that when we moved to Alloy we transformed some static components into empty shells: https://github.com/grafana/alloy/pull/55/files#diff-1b127540731f7403cbe5211de129eddfc35579d88d6bcbc371520642a9aa5486. The problem with this component is that the default values were set in the newRemoteWriteExporter() function. This StaleTime default value is actually supposed to be 15 minutes. I got rid of the check in the converter and set the correct value in the yaml component. This way, if you convert a config that uses the default yaml value, it will be correctly converted to 15m in Alloy.